rest/models/src/main/java/com/connexta/store/rest/models/AddMetadataRequest.java

```diff
@@ -0,0 +1,82 @@
+package com.connexta.store.rest.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+public class AddMetadataRequest {
+
+  @JsonProperty("metadataInfos")
+  @Valid
+  private List<MetadataInfo> metadataInfos = new ArrayList<>();
+
+  public AddMetadataRequest metadataInfos(List<MetadataInfo> metadataInfos) {
+    this.metadataInfos = metadataInfos;
+    return this;
+  }
+
+  public AddMetadataRequest addMetadataInfo(MetadataInfo metadataInfo) {
+    this.metadataInfos.add(metadataInfo);
+    return this;
+  }
+
+  /**
+   * An array of the information for each metadata type that will be added to the Dataset.
+   *
+   * @return metadataInfos
+   */
+  @ApiModelProperty(
+      required = true,
+      value = "An array of the information for each metadata type that will be added to the Dataset.")
+  @NotNull
+  @Valid
+  @Size
+  public List<MetadataInfo> getMetadataInfos() {
+    return List.copyOf(metadataInfos);
+  }
+
+  public void setMetadataInformations(List<MetadataInfo> metadataInfos) {
+    this.metadataInfos = metadataInfos;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
There is a Lombok annotation to create equals and hash methods. `@EqualsAndHashCode`. I haven't used it, but it probably works. I guess it depends on how do you feel about adding a Lombok dependency to an API project. I can certainly understand if you wouldn't want to.

Reply:
This class uses the OpenAPI spec generation as a guide. I kept it this way for consistency.